src/test/java/picard/sam/util/PrimaryAlignmentKeyTest.java

```diff
@@ -0,0 +1,114 @@
+package picard.sam.util;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.Tuple;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+public class PrimaryAlignmentKeyTest {
+
+    final SAMFileHeader samHeader = new SAMFileHeader();
+
+    @DataProvider(name="positiveTestData")
+    public Object[][] positiveTestData() {
+        return new Object[][]{
+                // unpaired < first
+                { makeRecordPair("r1", 0,
+                        "r1", SAMFlag.READ_PAIRED.intValue() | SAMFlag.FIRST_OF_PAIR.intValue()), -1 },
```

Comment:
It would be slightly cleaner to use a `Set` here, e.g. `makeRecordPair("r1", Collections.emptySet(), "r1", EnumSet.of(SAMFlag.READ_PAIRED, SAMFlag.FIRST_OF_PAIR))`
and below in `makeRecordPair()`
```
rec.setFlags(readFlags.stream().map(SAMFlag::intValue).reduce(0, (a, b) -> a | b));
```

Reply:
Done in previous round.